### PR TITLE
fix std::bad_alloc on pci_devices on Apple Silicon macs

### PR DESCRIPTION
**Pull request by sharvilshah**
Sometimes querying `pci_devices` on Apple Silicon macs would result in `std::bad_alloc`. 
The way to trigger this would be `select * from pci_devices` in quick succession (about 10ish times). This was because we were not doing bounds check on a vector

### CODE REVIEW

File: osquery/utils/conversions/darwin/iokit.cpp [hunk 1]
```diff
@@ -45,8 +45,11 @@ IOKitPCIProperties::IOKitPCIProperties(const std::string& compatible) {
 
   std::vector<std::string> vendor;
   boost::split(vendor, properties[prop_index++], boost::is_any_of(","));
+
   vendor_id = vendor[0].substr(3);
```
Comment (directionless) [inline osquery/utils/conversions/darwin/iokit.cpp:49]: Is this something that we need to check, or is this guaranteed by something earlier?

**Review by directionless — APPROVED** [review]
Seems better than it is today!

File: osquery/utils/conversions/darwin/iokit.cpp (hunk 1 above)
Comment (Smjert) [reply osquery/utils/conversions/darwin/iokit.cpp:49]: @sharvilshah Thanks for catching this!

I wouldn't depend on the property always containing at least a comma; better be safe than sorry and check that the `vendor` is not empty first.

Comment (sharvilshah) [reply osquery/utils/conversions/darwin/iokit.cpp]: @Smjert Thanks! Agree, let's do an extra check, updated!

**Review by Smjert — APPROVED** [review]

**Pull request merged**